.gitignore

```diff
@@ -71,3 +71,4 @@ cmake/.idea
 /.vs/SG14/v15/Browse.VC.opendb
 /.vs/SG14/v15/Browse.VC.db
 /.vs/ProjectSettings.json
+/[Bb]uild*/
```

Comment:
Could you reasonably restrict this to just `/Build/` and `/build/`? I mean, do you know of a build tool that would create `/build8624/` or `/Build_Foo/` that would need ignoring?

The commit message should explain why `/Build/` and `/build/` need ignoring (i.e., what build tools create them).

Reply:
Of course.

Reply:
Incidentally, what about the rest of the stuff in the removed `/cmake/.gitignore` — `/cmake/CMakeCache.txt`and such? I infer that the idea is that with a proper build process `CMakeCache.txt` will end up nested under the completely-ignored `/build`, not under `/cmake`, so `CMakeCache.txt` no longer needs to be explicitly ignored, is that right?

I'm skeptical of the ignoring of `/Build`, but maybe that matters on Windows? (But I thought Windows' filesystem might be case-insensitive anyway?)

Reply:
Correct, everything cmake generated will be in /build, so ignoring that folder is the norm. I've seen people use `Build`, but I don't mind removing it. It isn't a windows thing.

I'm thinking, maybe I should update the readme to indicate the change (`mkdir build && cd build` etc).

Reply:
> update the readme

Yeah, probably a good idea!  FYI, I always just `cd SG14_test && g++ -std=c++14 -DTEST_MAIN -I../SG14 whatever_test.cpp && ./a.out`; it might be worth mentioning that lazy method in the readme as well.
Describing how to use the cmake file is definitely a good idea, since my understanding starts getting hazy right about where you put the "etc". ;)  Just `mkdir build && cd build && cmake .. && ./test`, something like that?

Reply:
You can take a look. Nothing exhaustive but its always nice to have copy-paste ready instructions :)